src/v3/src/components/LoopbackProbe/LoopbackProbe.tsx

```diff
@@ -0,0 +1,172 @@
+};
+
+const LoopbackProbe: FunctionComponent<{ uischema: LoopbackProbeElement }> = ({
+  uischema: {
+    options: {
+      deviceChallengePayload,
+      cancelStep,
+      step,
+    },
+  },
+}) => {
+  const onSubmitHandler = useOnSubmit();
+
+  const probeTimeoutMillis: number = typeof deviceChallengePayload.probeTimeoutMillis === 'undefined'
+    ? 100 : deviceChallengePayload.probeTimeoutMillis;
+  const ports: number[] = deviceChallengePayload.ports || [];
+  const { domain } = deviceChallengePayload;
+  const { challengeRequest } = deviceChallengePayload;
+
+  const cancelHandler = (params?: Record<string, unknown> | undefined) => {
+    if (typeof cancelStep !== 'undefined') {
+      onSubmitHandler({
+        isActionStep: true,
+        step: cancelStep,
+        params,
+      });
+    }
+  };
+
+  useEffect(() => {
+    const doLoopback = async () => {
+      let foundPort = false;
+      // loop over each port
+      for (const port of ports) {
+        try {
+          // probe the port
+          const probeResponse = await makeRequest({
+            method: 'GET',
+            /*
+            OKTA-278573 in loopback server, SSL handshake sometimes takes more than 100ms and thus needs additional
+            timeout however, increasing timeout is a temporary solution since user will need to wait much longer in
+            worst case.
+            TODO: Android timeout is temporarily set to 3000ms and needs optimization post-Beta.
+            OKTA-365427 introduces probeTimeoutMillis; but we should also consider probeTimeoutMillisHTTPS for
+            customizing timeouts in the more costly Android and other (keyless) HTTPS scenarios.
+            */
+            timeout: isAndroid() ? 3_000 : probeTimeoutMillis,
+            url: `${domain}:${port}/probe`,
+          });
+
+          if (!probeResponse.ok) {
+            Logger.error(`Authenticator is not listening on port ${port}.`);
+            // there's more ports to try, continue with next port
+            continue;
+          }
+
+          // try port with challenge request
+          const challengeResponse = await makeRequest({
+            url: `${domain}:${port}/challenge`,
+            method: 'POST',
+            timeout: 300_000,
+            data: JSON.stringify({ challengeRequest }),
+          });
+
+          if (!challengeResponse.ok) {
+            // Windows and MacOS return status code 503 when
+            // there are multiple profiles on the device and
+            // the wrong OS profile responds to the challenge request
+            if (challengeResponse.status !== 503) {
+              // when challenge response with other error statuses,
+              // cancel polling and return
+              cancelHandler({
+                reason: 'OV_RETURNED_ERROR',
+                statusCode: challengeResponse.status,
+              });
+
+              break;
+            }
+            // no errors but this is not the port we're looking for
```

Comment:
⭐ You deserve an Oktappreciate for the Star wars reference here 😅

Reply:
Haha I'm glad you noticed